### DISCUSSION TITLE
napalm get_lldp_neighbors

### DISCUSSION THREAD

**Discussion by djohnnes** [netbox-community/netbox#9433] [Q&A]
Dear friends,
I am able to get the json response from the: GET /api/dcim/devices/774/napalm/?method=get_lldp_neighbors

How can I access this data inside  the template?
I am assuming this data is temporary available somewhere, can you help me please.

Thank you,

HTTP 200 OK
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "get_lldp_neighbors": {
        "Ethernet0": [
            {
                "hostname": "junos-unittest0",
                "port": "520"
            }
        ],
        "Ethernet1": [
            {
                "hostname": "junos-unittest1",
                "port": "521"
            }
        ],
        "Ethernet2": [
            {
                "hostname": "junos-unittest2",
                "port": "522"
            }
        ],
        "Ethernet3": [
            {
                "hostname": "junos-unittest3",
                "port": "523"
            }
        ],
        "Ethernet4": [
            {
                "hostname": "junos-unittest4",
                "port": "524"
            }
        ]
    }
}

**Comment by candlerb** [comment]
> How can I access this data inside the template?

Which template?

**Reply by djohnnes** [reply]
The: 'dcim/device/lldp_neighbors.html'

**Reply by candlerb** [reply]
So you mean, you're modifying Netbox itself?

As far as I can see, `netbox/templates/dcim/device/lldp_neighbors.html` doesn't have any access to that data.  All it does is load some Javascript (`lldp.js`, which is built from `netbox/project-static/src/device/lldp.ts`). The javascript then calls the API endpoint to get lldp neighbors, and populates the HTML dynamically from that.

**Reply by djohnnes** [reply]
Yes I am modifying the netbox to explore other possibilities.

Thank you very much for your clear feedback, it is very helpful for me.
